packages/keyring-eth-simple/src/simple-keyring-v2.test.ts

```diff
@@ -0,0 +1,1026 @@
+      expect(accounts).toHaveLength(2);
+
+      const account1 = accounts[1];
+      expect(account1).toBeDefined();
+      await wrapper.deleteAccount((account1 as KeyringAccount).id);
+      accounts = await wrapper.getAccounts();
+      expect(accounts).toHaveLength(1);
+
+      const account0 = accounts[0];
+      expect(account0).toBeDefined();
+      await wrapper.deleteAccount((account0 as KeyringAccount).id);
+      accounts = await wrapper.getAccounts();
+      expect(accounts).toHaveLength(0);
+    });
+
+    it('can re-create accounts after deleting all', async () => {
+      // Delete all accounts
+      let accounts = await wrapper.getAccounts();
+      expect(accounts).toHaveLength(3);
+
+      const accountToDelete2 = accounts[2];
+      const accountToDelete1 = accounts[1];
+      const accountToDelete0 = accounts[0];
+      expect(accountToDelete2).toBeDefined();
+      expect(accountToDelete1).toBeDefined();
+      expect(accountToDelete0).toBeDefined();
+
+      await wrapper.deleteAccount((accountToDelete2 as KeyringAccount).id);
+      await wrapper.deleteAccount((accountToDelete1 as KeyringAccount).id);
+      await wrapper.deleteAccount((accountToDelete0 as KeyringAccount).id);
+
+      accounts = await wrapper.getAccounts();
+      expect(accounts).toHaveLength(0);
+
+      // Now we can create accounts again
+      const result = await wrapper.createAccounts({
+        type: 'private-key:import',
+        accountType: EthAccountType.Eoa,
+        encoding: PrivateKeyEncoding.Hexadecimal,
+        privateKey: TEST_PRIVATE_KEY_1,
+      });
+
+      expect(result).toHaveLength(1);
+
+      const final = await wrapper.getAccounts();
+      expect(final).toHaveLength(1);
+    });
+
+    it('handles concurrent deletion with mutex', async () => {
+      const accounts = await wrapper.getAccounts();
+      const account1 = accounts[0];
+      const account2 = accounts[1];
+
+      expect(account1).toBeDefined();
+      expect(account2).toBeDefined();
+
+      // Attempt concurrent deletion (should work due to mutex)
+      const promises = [
+        wrapper.deleteAccount((account1 as KeyringAccount).id),
+        wrapper.deleteAccount((account2 as KeyringAccount).id),
+      ];
+
+      await Promise.all(promises);
+
+      const remaining = await wrapper.getAccounts();
+      expect(remaining).toHaveLength(1);
+    });
+  });
+
+  describe('exportAccount', () => {
+    beforeEach(async () => {
+      await wrapper.createAccounts({
+        type: 'private-key:import',
+        accountType: EthAccountType.Eoa,
+        encoding: PrivateKeyEncoding.Hexadecimal,
+        privateKey: TEST_PRIVATE_KEY_1,
+      });
+    });
+
+    it('exports an account as private key in hexadecimal encoding', async () => {
+      const accounts = await wrapper.getAccounts();
+      const account = accounts[0];
+
+      expect(account).toBeDefined();
+
+      const accountId = account?.id ?? '';
+      expect(accountId).not.toBe('');
```

Comment:
Nit: We could also re-use the same pattern than before:

```suggestion
      expect(account?.id).toBeDefined();
      const accountId = (account as KeyringAccount).id;
```